pkg/mvs/mvs.go

```diff
@@ -0,0 +1,38 @@
+package mvs
+
+import (
+	"fmt"
+
+	"golang.org/x/mod/module"
+)
+
+type ReqsMap map[module.Version][]module.Version
+
+func (r ReqsMap) Max(_, v1, v2 string) string {
+	if v1 == "none" || v2 == "" {
+		return v2
+	}
+	if v2 == "none" || v1 == "" {
+		return v1
+	}
+	if v1 < v2 {
+		return v2
+	}
+	return v1
+}
+
+func (r ReqsMap) Upgrade(m module.Version) (module.Version, error) {
+	panic("unimplemented")
+}
+
+func (r ReqsMap) Previous(m module.Version) (module.Version, error) {
+	panic("unimplemented")
```

Comment:
Perhaps we can break down the unfinished features into smaller granularity PR.

Reply:
indeed ! I will add the implementation of these in the upgrade/downgrade PR